packages/block-editor/src/components/inserter/hooks/use-insertion-point.js

```diff
@@ -53,40 +53,38 @@ function useInsertionPoint( {
 		( select ) => {
 			const {
 				getSelectedBlock: _getSelectedBlock,
+				getSelectedBlockClientId,
+				getBlockRootClientId,
 				getBlockIndex,
 				getBlockOrder,
-				getBlockInsertionPoint,
 			} = select( blockEditorStore );
-
-			let _destinationRootClientId, _destinationIndex;
-
-			if ( rootClientId || insertionIndex || clientId || isAppender ) {
-				// If any of these arguments are set, we're in "manual mode"
-				// meaning the insertion point is set by the caller.
-
-				_destinationRootClientId = rootClientId;
-
-				if ( insertionIndex ) {
-					// Insert into a specific index.
-					_destinationIndex = insertionIndex;
-				} else if ( clientId ) {
-					// Insert after a specific client ID.
-					_destinationIndex = getBlockIndex(
-						clientId,
-						_destinationRootClientId
-					);
-				} else {
-					// Insert at the end of the list.
-					_destinationIndex = getBlockOrder(
+			const selectedBlockClientId = getSelectedBlockClientId();
+
+			let _destinationRootClientId = rootClientId;
+			let _destinationIndex;
+
+			if ( insertionIndex ) {
+				// Insert into a specific index.
+				_destinationIndex = insertionIndex;
+			} else if ( clientId ) {
```

Comment:
I think at some point it would be nice to clean this up and remove `isAppender` and `clientID` in favour of `insertionIndex`.